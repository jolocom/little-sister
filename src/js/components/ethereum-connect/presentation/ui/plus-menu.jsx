import React from 'react'
import Radium from 'radium'
import {theme} from 'styles'
import {
  NavigationExpandMore,
  NavigationExpandLess
} from 'material-ui/svg-icons'
import { List, FloatingActionButton, Divider } from 'material-ui'
import {IconCheckmark} from '../../../common'

const STYLES = {
  addBtn: {
    position: 'absolute',
    top: '12.5px',
    right: '8px'
  },
  accordionBtn: {
    position: 'absolute',
    top: '12.5px',
    right: '56px'
  },
  iconCreate: {
    fill: theme.palette.accent1Color
  },

  infoHeader: {
    textAlign: 'left',
    color: theme.textStyles.sectionheader.color,
    fontSize: theme.textStyles.sectionheader.fontSize,
    fontWeight: theme.textStyles.sectionheader.fontWeight,
    marginBottom: '15px',
    display: 'inline-block'
  },
  item: {
    alignItems: 'center',
    padding: '0 0 0 36px',
    display: 'inline-block',
    verticalAlign: 'top',
    width: '100%',
    boxSizing: 'border-box'
  },
  root: {
    position: 'relative',
    width: '100%',
    whiteSpace: 'nowrap'
  },
  securityLevel: {
    paddingLeft: '8px',
    display: 'inline-flex',
    flexDirection: 'row',
    alignItems: 'center'
  },
  itemSecurity: {
    order: 1,
    padding: '8px'
  }
}
const PlusMenu = (props) => {
  const icon = props.expanded ? <NavigationExpandLess />
    : <NavigationExpandMore />
  const renderSecurityIcons = props.securityDetails.map((item, index) => {
    return (
      <IconCheckmark
        key={index}
        style={STYLES.itemSecurity}
        color={item.verified ? theme.palette.primary1Color
          : theme.jolocom.gray1} />
    )
  })
  return (
    <div style={Object.assign(STYLES.root, props.style)}>
      <List>
        <div style={STYLES.item}>
          <div style={STYLES.infoHeader}>
          {props.name}
            <div style={STYLES.securityLevel}>
              {renderSecurityIcons}
            </div>
          </div>
          <Divider style={STYLES.divider} />
        </div>
        <div style={STYLES.addBtn}>
          <FloatingActionButton
            mini
            secondary={!props.choice}
            onClick={() => props.toggle(!props.expanded)}
            containerElement="label"
            style={STYLES.addBtn}
            backgroundColor={'#FFF'}
            iconStyle={STYLES.iconCreate}>
            {icon}
          </FloatingActionButton>
        </div>
      </List>
      {props.children}
    </div>
  )
}

PlusMenu.propTypes = {
  name: React.PropTypes.any,
  children: React.PropTypes.node,
  style: React.PropTypes.object,
  expand: React.PropTypes.func,
  expanded: React.PropTypes.bool,
  choice: React.PropTypes.bool
}

export default Radium(PlusMenu)
