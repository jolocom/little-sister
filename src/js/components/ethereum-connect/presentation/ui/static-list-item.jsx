import React from 'react'
import Radium from 'radium'

import {theme} from 'styles'
import {
  ListItem
} from 'material-ui'

const STYLES = {
  iconVerified: {
    color: theme.palette.primary1Color,
    top: '20px'
  },
  icon: {
    color: theme.jolocom.gray1,
    top: '20px'
  },
  iconAvatar: {
    backgroundColor: 'none',
    borderRadius: '0%',
    top: '16px'
  },
  unverifiedListItem: {
    paddingBottom: '0px'
  },
  listItem: {
    padding: '0 16px 0 36px'
  },
  text: {
    marginRight: '8px'
  },
  textSecurity: {
    fontSize: '14px'
  }
}

@Radium
export default class StaticListItem extends React.Component {
  static propTypes = {
    icon: React.PropTypes.any,
    verified: React.PropTypes.bool,
    secondaryTextValue: React.PropTypes.string
  }

  render() {
    const props = this.props
    const icon = props.icon
      ? <props.icon color={props.verified ? STYLES.iconVerified.color
      : STYLES.icon.color} style={{top: '10px'}} /> : <div />

    return (
      <div style={STYLES.listItem}>
        <ListItem
          innerDivStyle={STYLES.text}
          primaryText={<p style={STYLES.textSecurity}>{props.type}</p>}
          secondaryTextLines={2}
          secondaryText={<div>
            <p>{props.text}</p></div>}
          leftIcon={icon}
          disabled />
      </div>
    )
  }
}
