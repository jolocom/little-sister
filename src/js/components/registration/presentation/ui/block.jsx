import React from 'react'
import Radium from 'radium'

const STYLE = {
  padding: '8px 0'
}

const Block = ({style, children, ...props}) => {
  return (
    <div style={Object.assign({}, STYLE, style)} {...props}>
      {children}
    </div>
  )
}

Block.propTypes = {
  children: React.PropTypes.node,
  style: React.PropTypes.object
}

export default Radium(Block)
