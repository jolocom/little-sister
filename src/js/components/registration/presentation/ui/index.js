export { default as Container } from './container'
export { default as Header } from './header'
export { default as Block } from './block'
export { default as Content } from './content'
export { default as Footer } from './footer'
export { default as SideNote } from './side-note'
