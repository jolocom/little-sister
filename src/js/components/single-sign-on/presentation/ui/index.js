export { default as SubMenuIcon } from './sub-menu-icon'
export { default as AddServiceIcon } from './add-service-icon'
export { default as ServiceIcon } from './service-icon'
