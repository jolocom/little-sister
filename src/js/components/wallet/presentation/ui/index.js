
export { default as EditHeader } from './edit-header'
export { default as EditAppBar } from './edit-app-bar'
export { default as EditListItem } from './edit-list-item'
export { default as TabContainer } from './tab-container'
export { default as PlusMenu } from './plus-menu'
export { default as PlusSubMenu } from './plus-sub-menu'
export { default as AddNew } from './add-new'
export { default as VerifiedShield } from './verified-shield'
export { default as StaticListItem } from './static-list-item'
export { default as HalfScreenContainer } from './half-screen-container'
export { default as SelectCountryItem } from './select-country-item'
export { default as SelectListItem } from './select-list-item'
export { default as DateListItem } from './date-list-item'
export { default as IconNumber } from './numbers-icon'
export { default as Bubbles } from './bubbles'
export { default as ContactList } from './contact-list-items'
export { default as IdCardsList } from './id-cards-list'
export { default as PassportsList } from './passports-list'
export { default as InfoDetails } from './info-details'
export { default as SmsCodeInput } from './sms-code-input'
export { default as VerificationButtons } from './verification-buttons'
export { default as VerificationButtonMsg } from './verification-button-msg'
export { default as InputMsg } from './sms-input-msg'
