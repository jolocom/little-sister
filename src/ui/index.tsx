export { Loading } from './registration/containers/loading'
