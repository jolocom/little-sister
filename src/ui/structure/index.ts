export { Wrapper } from './wrapper'
export { CardWrapper } from './cardWrapper'
export { JolocomButton, OnlineJolocomButton } from './jolocomButton'
