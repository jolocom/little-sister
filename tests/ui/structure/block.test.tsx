// TODO UPDATE WITH ENZYME
import React from 'react'
import { Block } from 'src/ui/structure/'
import { StyleSheet } from 'react-native'

const ShallowRenderer = require('react-test-renderer/shallow')

describe('generic block component', () => {
  it('matches the snapshot with no extra props', () => {
    const renderer = new ShallowRenderer()
    const rendered = renderer.render(<Block> null </Block>)
    expect(rendered).toMatchSnapshot()
  })

  it('matches snapshot with flex prop', () => {
    const renderer = new ShallowRenderer()
    const rendered = renderer.render(<Block flex={0.1}> null </Block>)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with debug prop', () => {
    const renderer = new ShallowRenderer()
    const rendered = renderer.render(<Block debug> null </Block>)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with style prop', () => {
    const renderer = new ShallowRenderer()
    const styles = StyleSheet.create({
      block: {
        width: '30%',
        backgroundColor: 'blue',
      },
    })

    const rendered = renderer.render(
      <Block style={styles.block} onTouch={() => {}}>
        null
      </Block>,
    )
    expect(rendered).toMatchSnapshot()
  })
})
