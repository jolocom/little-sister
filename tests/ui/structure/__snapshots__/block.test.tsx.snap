// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic block component matches snapshot with flex prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
      Object {
        "flex": 0.1,
      },
    ]
  }
>
   null 
</View>
`;

exports[`generic block component matches the snapshot with debug prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
      Object {
        "borderColor": "#d6d7da",
        "borderRadius": 4,
        "borderWidth": 0.5,
      },
    ]
  }
>
   null 
</View>
`;

exports[`generic block component matches the snapshot with no extra props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
   null 
</View>
`;

exports[`generic block component matches the snapshot with style prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      },
      Object {
        "backgroundColor": "blue",
        "width": "30%",
      },
    ]
  }
>
  null
</View>
`;
